### ISSUE TITLE
Reactions container stretches beyond width of comment area

### ISSUE THREAD

**Issue by Daniel15** [sindresorhus/refined-github#1744]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2430%20Funded-%2300A156.svg" />](https://issuehunt.io/r/sindresorhus/refined-github/issues/1744)

The new "eyeballs" and "spaceship" reactions are making the container too wide

![](https://d.sb/2019/01/firefox_24-12.05.36.png)

See how it overlaps the right column.

PR in the screenshot: https://github.com/facebook/react/pull/14679

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $30.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($30.00)
### Submitted pull Requests
- [#2416 Wrap reactions-avatars](https://issuehunt.io/r/sindresorhus/refined-github/pull/2416)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/refined-github/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

**Comment by bfred-it** [comment]
`if types > 4, return`

We can't add any avatars in that case, unless we stop at 1/2 avatars, which would look kinda wonky

<img width="710" alt="screenshot 2019-01-27 at 21 12 01" src="https://user-images.githubusercontent.com/1402241/51802862-40cb5c80-2278-11e9-8d84-8ec188ab8007.png">

**Comment by jerone** [comment]
What about some flexbox...
![image](https://user-images.githubusercontent.com/55841/51805933-e6300180-2273-11e9-9d63-68de559f2be0.png)

```css
.comment-reactions-options {
    flex-wrap: wrap;
    justify-content: stretch;
    margin-bottom: -1px;
}
.reaction-summary-item {
    border-bottom: 1px solid #d1d5da;
    flex: 1 1 auto;
}
```

**Comment by bfred-it** [comment]
By using grid we can probably keep the width constant and show more avatars… but it’s a lot of noise and height 🤔

**Comment by gurrrung** [comment]
@bfred-it thoughts about the @jerone suggestion? It looks pretty smooth.

**Comment by bfred-it** [comment]
I don’t think it will look great with 6 reactions that way: 5 on the first line and 1 stretched out on the second line

**Comment by bfred-it** [comment]
@jerone do you want to take this one?

Further changes needed since we now allow 2 lines:

- show more avatars or all of them
- drop the "expand on hover" behavior because it makes things harder; just always display the avatars expanded
- manually wrap the reactions to make them balanced (so the second line never ends up having only 1 reaction with 1 avatar, for example); this might not be easy because reaction count also applies, but perhaps this logic should be good enough:
	- if it wraps, 
		- 4 reactions: 2 + 2
		- 5 reactions: 3 + 2
		- 6 reactions: 3 + 3
		- 7 reactions: 4 + 3
		- 8 reactions: 4 + 4

**Comment by jerone** [comment]
@fregante For the next 1.5 week my time is sparse, but after then, yes I'll send a PR. In the meantime, if someone else wants to that's fine too; we all use this tool :)

**Comment by fregante** [comment]
Ping on this if anyone pick it up

**Comment by jerone** [comment]
Yeah, was already pocking at it.

**Comment by jerone** [comment]
Work-in-progress, but this is what I have so far:

![image](https://user-images.githubusercontent.com/55841/64479598-ab9f2780-d1b9-11e9-93d1-dd05be8d13bb.png)

```
.has-reactions form {
	width: 100%;
}

.rgh-reactions {
	display: grid;
	grid-template-columns: repeat(8, 25%);
	width: 100%;
	justify-content: stretch;
}

.comment-reactions-options {
	margin-bottom: -1px;
	background: transparent;
}
.reaction-summary-item {
	border-bottom: 1px solid #d1d5da;
}
.reaction-popover-container .reaction-summary-item {
	border-bottom: 0 none; /* No border below the "Add your reaction" button. */
}
```

**Comment by ulken** [comment]
@fregante What's the status of this?

If no one is actively working on it, I'd like to take a stab at it.

**Comment by jerone** [comment]
See https://github.com/sindresorhus/refined-github/pull/2416

**Comment by jerone** [comment]
@fregante I'll might take another stab at this soon. Can you summarize for the last time what kind of wrapping you want.

**Comment by fregante** [comment]
Thank you!

Logic: only enable the 2-line grid when the icons don't fit on one line.

The easiest solution (without math guesses) would be to measure that with `getBoundingClientRect` (if height > 50, enable grid) but each call triggers a reflow, so you'd have to batch them. e.g:

```js
for (const line of reactionLines.map(el => ({el, height: el.getBoundingClientRect().height}))) {
	if (height > 50) {
		line.classList.add('grid')
	}
}
```
